doc/Programming-guide/Key-platform-features/Endpoint-profiles/index.md

```diff
@@ -6,117 +6,438 @@ nav: /:path/Programming-guide/Key-platform-features/Endpoint-profiles/
+groups membership.
+
+Think about the client-side profile schema as of a structured data set of your endpoint application that will later be available to you in Kaa server and may change 
+due to your client application logic or device state.
+You can configure your own client-side profile schema using the 
+[Admin UI](#setting-client-side-endpoint-profile-update-from-admin-ui) or [REST API]({{root_url}}Programming-guide/Server-REST-APIs). 
+For the purpose of this guide we will use a fairly abstract client-side profile schema shown below.
+
+```json{
+    "type":"record",
+    "name":"Profile",
+    "namespace":"org.kaaproject.kaa.schema.sample.profile",
+    "fields":[
+        {
+            "name":"id",
+            "type":"string"
+        },
+        {
+            "name":"os",
+            "type":{
+                "type":"enum",
+                "name":"OS",
+                "symbols":[
+                    "Android",
+                    "iOS",
+                    "Linux"
+                ]
+            }
+        },
+        {
+            "name":"os_version",
+            "type":"string"
+        },
         {
-            "name": "hobbies",
-            "type": { "type": "array", "items": "string" }
+            "name":"build",
+            "type":"string"
         }
     ]
 }
 ```
 
-## The following client-side profile would be compatible with our schema example. ##
+Client-side endpoint profile updates are reported to the endpoint SDK using a profile container. The profile related API varies depending on the target SDK platform, 
+however the general approach is the same.
+
+<ul class="nav nav-tabs">
+  <li class="active"><a data-toggle="tab" href="#Java">Java</a></li>
+  <li><a data-toggle="tab" href="#C_plus_plus">C++</a></li>
+  <li><a data-toggle="tab" href="#C">C</a></li>
+  <li><a data-toggle="tab" href="#Objective-C">Objective-C</a></li>
+</ul>
+
+<div class="tab-content">
+<div id="Java" class="tab-pane fade in active" markdown="1" >
+
+```java
+import org.kaaproject.kaa.client.Kaa;
+import org.kaaproject.kaa.client.KaaClient;
+import org.kaaproject.kaa.client.profile.ProfileContainer;
+import org.kaaproject.kaa.client.DesktopKaaPlatformContext;
+import org.kaaproject.kaa.schema.sample.profile.OS;
+import org.kaaproject.kaa.schema.sample.profile.Profile;
+ 
+// Profile is an auto-generated class based on user defined schema.
+Profile profile;
+// Desktop Kaa client initialization
+public void init() {
+    // Create instance of desktop Kaa client for Kaa SDK
+    KaaClient client = Kaa.newClient(new DesktopKaaPlatformContext(),
+            new SimpleKaaClientStateListener());
+    // Sample profile
+    profile = new Profile("id", OS.Linux, "3.17", "0.0.1-SNAPSHOT");
+    // Simple implementation of ProfileContainer interface that is provided by the SDK
+    client.setProfileContainer(new ProfileContainer() {
```

Comment:
In java-client-core we don't have any class or interface with name DefaultProfileContainer